plugins/document/dynamodb/dynamo_db_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamodb_service_test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	ds_plugin "github.com/nitric-dev/membrane/plugins/document/dynamodb"
+	"github.com/nitric-dev/membrane/plugins/document/test"
+	"github.com/onsi/ginkgo"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+func startDynamoProcess() *exec.Cmd {
+	// Start Local DynamoDB
+	args := []string{
+		"-Djava.library.path=/usr/local/dynamodb/DynamoDBLocal_lib",
+		"-jar",
+		"/usr/local/dynamodb/DynamoDBLocal.jar",
+		"-inMemory",
+	}
+	cmd := exec.Command("/usr/bin/java", args[:]...)
+	if err := cmd.Start(); err != nil {
+		panic(fmt.Sprintf("Error starting Local DynamoDB %v : %v", cmd, err))
+	}
+	fmt.Printf("Started Local DynamoDB (PID %v) and loading data...\n", cmd.Process.Pid)
+
+	return cmd
+}
```

Comment:
I would like to discuss our testing strategy for integrations vs. unit tests. These tests are great, but I would like to separate them out into their own suite and make target.

Something like:
```makefile
# Pure unit tests
test-unit:
  ... run unit tests
 
# Integration tests
test-integration:
  ... run integration tests (this includes dependencies on external software)

# Aggregate test make target.
test: test-unit, test-integration
 @echo running tests
```
Also it would be ideal to containerise the local integration test suite and provide a either a pre-built image or an in repo dockerfile that is prepared inline to simplify the process. Running the tests is one of the first things potential contributors will probably do and lowering potential for failure and simplifying setup would help us lower rejection.

It could also be handy to add a timeout for starting the external processes as well. Currently the test suite just hangs, on a machine without the required files.